### ISSUE TITLE
Address or Remove all TODO comments

### ISSUE THREAD

**Issue by vkurup** [unicef/etools-issues#640]
Leaving TODO comments in the codebase causes confusion. We should remove them, either fixing the
issue, if we can identify what it is, or creating a Waffle issue for others to determine if it is
still relevant.

**Comment by domdinicola** [comment]
can you start having a look to this?

**Comment by vkurup** [comment]
@domdinicola Yes I will. I'd like to get something reviewable for #578 first and then I'll tackle this.
